### PR TITLE
*(ticdc): optimize memory usage of RowChangedEvent

### LINKED ISSUE #10386: Optimize memory usage of TiCDC

### PR DESCRIPTION
**Pull request by lidezhu**
### What problem does this PR solve?

Issue Number: close #10386 

### What is changed and how it works?
1. Add a more memory efficient struct `ColumnData` to represent column data, and the schema info of column can only be fetched from `TableInfo`;
2. Replace `[]*Column` in `RowChangedEvent` with `[]*ColumnData`;
3. Add method `RowChangedEvent.GetColumns` and `RowChangedEvent.GetPreColumns` to get the `Column` representation of columns;
4. There should be no `nil` in `RowChangedEvent.Columns` and `RowChangedEvent.PreColumns`, so this pr remove some code in sink module to handle `nil` value.
5. Maintain a correct `TableInfo` in all kinds of tests;

#### Tests 

 - Unit test
 - Manual test (add detailed scripts or steps below)
1. disable tidb gc;
6. load 10M rows of a wide table;
7. create changefeed with `memory-quota` set 8GB;
8. compare the peak memory usage;
Before optimization, peak memory usage: 26.5GB,
image
After optimization, peak memory usage: 10.2GB,
image
The peak memory usage decrease about 60%.

### Release note 

```release-note
None.
```

### CODE REVIEW

File: cdc/entry/mounter_test.go [hunk 1]
```diff
@@ -1641,7 +1641,12 @@ func TestBuildTableInfo(t *testing.T) {
 		originTI, err := ddl.BuildTableInfoFromAST(stmt.(*ast.CreateTableStmt))
 		require.NoError(t, err)
 		cdcTableInfo := model.WrapTableInfo(0, "test", 0, originTI)
-		cols, _, _, _, err := datum2Column(cdcTableInfo, map[int64]types.Datum{})
```
Comment (asddongmen) [inline cdc/entry/mounter_test.go:1644]: This err should also be checked.

**Review by asddongmen — APPROVED** [review]

**Comment by asddongmen** [comment]
Excellent work! Respect!

File: cdc/entry/mounter_test.go (hunk 1 above)
Comment (lidezhu) [reply cdc/entry/mounter_test.go:1644]: Fixed.

**Review by sdojjy — APPROVED** [review]

**Pull request merged**